src/lib.rs

```diff
@@ -1,5 +1,268 @@
+    /// `Option::take`.
+    ///
+    /// This method moves the specified path to the intended output file out of
+    /// this struct. This should, hence, be called just once but prevents an
+    /// obsolete clone.
+    #[must_use]
+    pub fn take_output(&mut self) -> Option<PathBuf> {
+        self.output.take()
+    }
+
+    /// Creates a new Filter as specified by the user.
+    #[must_use]
+    pub fn filter(&self) -> Filter {
+        Filter {
+            author_contains: &self.author_contains,
+            author_equals: &self.author_equals,
+            commit_contains: &self.commit_contains,
+            commit_equals: &self.commit_equals,
+            email_contains: &self.email_contains,
+            email_equals: &self.email_equals,
+            file_extension: &self.file_extension,
+            message_contains: &self.message_contains,
+            message_equals: &self.message_equals,
+            message_starts_with: &self.message_starts_with,
+        }
+    }
+}
+
+/// The possible input methods.
+#[derive(Subcommand, Debug)]
+pub enum InputMethod {
+    /// Read the input from the local Git history.
+    GitHistory,
+
+    /// Read the specified input file.
+    LogFile {
+        /// The log file to read from.
+        log_file: PathBuf,
+    },
+
+    /// Read from `stdin`.
+    Stdin,
+}
+
+impl InputMethod {
+    /// Process the configured input method.
+    pub fn read(&self) -> Result<String, Box<dyn std::error::Error>> {
+        match self {
+            Self::GitHistory => {
+                let process = std::process::Command::new("git")
+                    .arg("log")
+                    .arg("--numstat")
+                    .output()?;
+
+                Ok(String::from_utf8(process.stdout)?)
+            }
+            Self::LogFile { log_file } => Ok(std::fs::read_to_string(log_file)?),
+            Self::Stdin => {
+                let mut input = String::new();
+
+                loop {
+                    let mut buffer = String::new();
+
+                    if std::io::stdin().read_line(&mut buffer)? == 0 {
+                        break;
+                    }
+
+                    input.push_str(&buffer);
+                }
+
+                Ok(input)
+            }
+        }
+    }
+}
+
+/// The revealed filter creteria.
+///
+/// This data structure allows to filter the input commits by certain criteria.
+#[derive(Debug, Default)]
+pub struct Filter<'a> {
+    /// A set of substrings to be contained by some authors' names.
+    author_contains: &'a [String],
+
+    /// A set of strings to match some authors's names.
+    author_equals: &'a [String],
+
+    /// A set of substrings to be contained by some commits' hashes.
+    commit_contains: &'a [String],
+
+    /// A set of strings to match some commits' hashes.
+    commit_equals: &'a [String],
+
+    /// A set of substrings to be contained by some authors' email
+    /// addresses.
```

Comment:
Why two lines?  Your docstrings above even longer and with the word "addresses" on the same line, you do not even exceed 80 characters!